File: core/static/core/js/date_utils.js
/**
 * date_utils.js
 * Dinamik tarih seçenekleri ve hesaplama — ortak kullanım için.
 * SAP tarih formatı: DD.MM.YYYY
 */

const SAP_DATE_OPTIONS = [
    { value: '',                label: '— Seçiniz —',          disabled: true },
    { value: 'today',           label: 'Bugün' },
    { value: 'year_start',      label: 'Bu yılın başı' },
    { value: 'month_start',     label: 'Bu ayın başı' },
    { value: 'year_end',        label: 'Bu yılın sonu' },
    { value: 'month_end',       label: 'Bu ayın sonu' },
    { value: 'prev_month_start',label: 'Önceki ayın başı' },
    { value: 'prev_month_end',  label: 'Önceki ayın sonu' },
    { value: 'month_5',         label: 'Bu ayın 5\'i' },
    { value: 'prev_year_start', label: 'Geçen yılın başı' },
    { value: 'days_15',         label: '15 gün önce' },
    { value: 'days_30',         label: '30 gün önce' },
    { value: 'days_45',         label: '45 gün önce' },
    { value: 'days_60',         label: '60 gün önce' },
    { value: 'days_365',        label: '365 gün önce' },
];

/**
 * Verilen anahtar için bugüne göre tarih hesaplar.
 * @param {string} key – SAP_DATE_OPTIONS içindeki value
 * @returns {string} DD.MM.YYYY formatında tarih veya ''
 */
function calcDynamicDate(key) {
    const now  = new Date();
    const y    = now.getFullYear();
    const m    = now.getMonth(); // 0-based
    let d;

    switch (key) {
        case 'today':           d = new Date(y, m, now.getDate()); break;
        case 'year_start':       d = new Date(y, 0, 1); break;
        case 'month_start':      d = new Date(y, m, 1); break;
        case 'year_end':         d = new Date(y, 11, 31); break;
        case 'month_end':        d = new Date(y, m + 1, 0); break;
        case 'prev_month_start': d = new Date(y, m - 1, 1); break;
        case 'prev_month_end':   d = new Date(y, m, 0); break;
        case 'month_5':          d = new Date(y, m, 5); break;
        case 'prev_year_start':  d = new Date(y - 1, 0, 1); break;
        case 'days_15':          d = new Date(now - 15  * 86400000); break;
        case 'days_30':          d = new Date(now - 30  * 86400000); break;
        case 'days_45':          d = new Date(now - 45  * 86400000); break;
        case 'days_60':          d = new Date(now - 60  * 86400000); break;
        case 'days_365':         d = new Date(now - 365 * 86400000); break;
        default: return '';
    }

    const dd   = String(d.getDate()).padStart(2, '0');
    const mm   = String(d.getMonth() + 1).padStart(2, '0');
    const yyyy = d.getFullYear();
    return `${dd}.${mm}.${yyyy}`;
}

File: core/utils/date_utils.py
"""Dynamic date helpers used by SAP runtime placeholders.

Originally lived in ``core/views.py``.
"""
from __future__ import annotations


def _calc_dynamic_date(key):
	"""JS calcDynamicDate'in Python karşılığı — SAP DD.MM.YYYY formatında tarih döner."""
	from datetime import date, timedelta
	import calendar
	today = date.today()
	y, m = today.year, today.month

	if key == 'today':
		d = today
	elif key == 'yesterday':
		d = today - timedelta(days=1)
	elif key == 'year_start':
		d = date(y, 1, 1)
	elif key == 'month_start':
		d = date(y, m, 1)
	elif key == 'year_end':
		d = date(y, 12, 31)
	elif key == 'month_end':
		d = date(y, m, calendar.monthrange(y, m)[1])
	elif key == 'prev_month_start':
		pm = m - 1 if m > 1 else 12
		py = y if m > 1 else y - 1
		d = date(py, pm, 1)
	elif key == 'prev_month_end':
		pm = m - 1 if m > 1 else 12
		py = y if m > 1 else y - 1
		d = date(py, pm, calendar.monthrange(py, pm)[1])
	elif key == 'month_5':
		d = date(y, m, 5)
	elif key == 'prev_year_start':
		d = date(y - 1, 1, 1)
	elif key == 'days_15':
		d = today - timedelta(days=15)
	elif key == 'days_30':
		d = today - timedelta(days=30)
	elif key == 'days_45':
		d = today - timedelta(days=45)
	elif key == 'days_60':
		d = today - timedelta(days=60)
	elif key == 'days_365':
		d = today - timedelta(days=365)
	else:
		return ''

	return f"{d.day:02d}.{d.month:02d}.{d.year}"
